### PR TITLE
Enhance avatar urls on upload (encode url) #368

### PR DESCRIPTION
**Pull request by ahsanbagwan**
Changes:
- JPA entity and repository
- Avatar Service layer
- Properties file
- Test files
- Test schema

### CODE REVIEW

File: src/main/java/com/faforever/api/data/domain/Avatar.java [hunk 1]
```diff
@@ -5,6 +5,7 @@
 import com.yahoo.elide.annotation.Include;
 import com.yahoo.elide.annotation.UpdatePermission;
 import lombok.Setter;
+import org.springframework.data.annotation.Transient;
```
Comment (bukajsytlos) [inline src/main/java/com/faforever/api/data/domain/Avatar.java]: javax.persistence.Transient is the correct one

File: src/main/java/com/faforever/api/avatar/AvatarRepository.java [hunk 2]
```diff
@@ -8,7 +8,8 @@
 
 @Repository
 public interface AvatarRepository extends JpaRepository<Avatar, Integer> {
-  Optional<Avatar> findOneByUrl(String url);
+  //Optional<Avatar> findOneByUrl(String url);
```
Comment (bukajsytlos) [inline src/main/java/com/faforever/api/avatar/AvatarRepository.java]: we don't leave any commented code

File: src/main/java/com/faforever/api/avatar/AvatarService.java [hunk 3]
```diff
@@ -47,12 +49,15 @@ public AvatarService(AvatarRepository avatarRepository, FafApiProperties propert
   public void createAvatar(AvatarMetadata avatarMetadata, String originalFilename, InputStream imageDataInputStream, long avatarImageFileSize) {
     final Avatar avatarToCreate = new Avatar();
     final String normalizedAvatarFileName = NameUtil.normalizeFileName(originalFilename);
-    String url = String.format(properties.getAvatar().getDownloadUrlFormat(), normalizedAvatarFileName);
-    avatarRepository.findOneByUrl(url).ifPresent(existingAvatar -> {
+    String url = String.format(properties.getAvatar().getBaseUrl(), normalizedAvatarFileName);
```
Comment (bukajsytlos) [inline src/main/java/com/faforever/api/avatar/AvatarService.java]: I not sure if this should be here. This is only during creation, but we should compute this attribute at read time also. see comment below

File: src/main/java/com/faforever/api/data/domain/Avatar.java [hunk 4]
```diff
@@ -22,14 +23,19 @@
 
   private String url;
   private String tooltip;
+  private String filename;
   private List<AvatarAssignment> assignments;
 
-  @Column(name = "url")
+  @Transient
   @NotNull
   public String getUrl() {
```
Comment (bukajsytlos) [inline src/main/java/com/faforever/api/data/domain/Avatar.java:39]: I think this should be made @ComputedAttribute and being enriched by EntityListener. Take a look at Clan class and ClanEnricherListener

**Review by bukajsytlos — CHANGES_REQUESTED** [review]

**Review by bukajsytlos — CHANGES_REQUESTED** [review]
You should also rebase this on (latest) develop branch

File: src/inttest/resources/sql/prepAvatarData.sql [hunk 5]
```diff
@@ -1,19 +1,25 @@
...
+DELETE
+FROM avatars;
+DELETE
+FROM avatars_list;
 DELETE
 FROM group_permission;
 DELETE
 FROM group_permission_assignment;
 
+SET FOREIGN_KEY_CHECKS = 1;
```
Comment (norraxx) [inline src/inttest/resources/sql/prepAvatarData.sql:12]: :see_no_evil: Truncate didn't work?

Comment (Brutus5000) [reply src/inttest/resources/sql/prepAvatarData.sql:12]: I'm tired figuring out the right order.

**Review by bukajsytlos — APPROVED** [review]

**Review by Brutus5000 — APPROVED** [review]

**Pull request merged**